moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -430,6 +460,32 @@ moveit_msgs::msg::MoveItErrorCodes plan_execution::PlanExecution::executeAndMoni
   path_became_invalid_ = false;
   bool preempt_requested = false;
 
+  // Check that attached objects remain consistent throughout the trajectory and store them.
+  // This avoids querying the scene for attached objects at each waypoint whenever possible.
+  // If a change in attached objects is detected, they will be queried at each waypoint.
+  plan_components_attached_objects_.clear();
+  plan_components_attached_objects_.reserve(plan.plan_components.size());
+  for (const auto& component : plan.plan_components)
+  {
+    const auto& trajectory = component.trajectory;
+    std::map<std::string, const moveit::core::AttachedBody*> trajectory_attached_objects;
+    if (trajectory)
+    {
+      std::map<std::string, const moveit::core::AttachedBody*> attached_objects;
+      trajectory->getWayPoint(0).getAttachedBodies(trajectory_attached_objects);
```

Comment:
This crashes with a segfault if trajectory is empty!
https://github.com/moveit/moveit_task_constructor/issues/692